src/platform/MovingPlatform.cpp

```diff
@@ -6,106 +6,122 @@
+  Rotation curr_r = getAttitude();
+  Rotation r = curr_r.applyTo(Rotation(Directions::up, angle));
+  if (angle > -eps && angle < eps) {
+    setAttitude(r);
+    return;
+  }
+  setAttitude(r);
+  angle = glm::angle(glm::normalize(cached_vectorToTarget_xy),
+                     cached_dir_current_xy);
+  if (angle > eps) { // If direct computation failed
+    std::stringstream ss;
+    ss << "It was not possible to determine attitude with a single "
+       << "computation at MovingPlatform::initLegManual\n\t"
+       << "angle = " << angle << " but it should be below " << eps
+       << "\n\tUsing iterative computation instead";
+    logging::WARN(ss.str());
+    initLegManualIterative();
+  }
 }
 
-void MovingPlatform::initLegManualIterative(){
-    try {
-        double angle = glm::angle(
-            glm::normalize(cached_vectorToTarget_xy),
-            glm::normalize(cached_dir_current_xy)
-        );
-        double stepSize = 0.025;
-        double heading_rad = 0;
-        while (angle > stepSize) {
-            heading_rad += angle;
-            Rotation r = Rotation(Directions::up, heading_rad);
-            this->setAttitude(r);
-            angle = glm::angle(
-                glm::normalize(cached_vectorToTarget_xy),
-                glm::normalize(cached_dir_current_xy)
-            );
-        }
-        logging::INFO(
-            "Iterative mode was used for manual leg initialization "
-            "because default one failed for MovingPlatform"
-        );
-    }
-    catch (std::exception &e) {
-        logging::WARN(e.what());
+void MovingPlatform::initLegManualIterative() {
+  try {
+    double angle = glm::angle(glm::normalize(cached_vectorToTarget_xy),
+                              glm::normalize(cached_dir_current_xy));
+    double stepSize = 0.025;
+    double heading_rad = 0;
+    while (angle > stepSize) {
+      heading_rad += angle;
+      Rotation r = Rotation(Directions::up, heading_rad);
+      this->setAttitude(r);
+      angle = glm::angle(glm::normalize(cached_vectorToTarget_xy),
+                         glm::normalize(cached_dir_current_xy));
     }
+    logging::INFO("Iterative mode was used for manual leg initialization "
+                  "because default one failed for MovingPlatform");
+  } catch (std::exception &e) {
+    logging::WARN(e.what());
+  }
 }
 
 bool MovingPlatform::waypointReached() {
-	// velocity is in m/cycle
-	// m / (m/cycle) => cycles left to reach waypoint
-	bool result = (glm::l2Norm(cached_vectorToTarget) / glm::l2Norm(velocity)) < 1.0;
-	if (result) logging::INFO("Waypoint reached!");
-	return result;
+  // velocity is in m/cycle
+  // m / (m/cycle) => cycles left to reach waypoint
+  bool result =
+      (glm::l2Norm(cached_vectorToTarget) / glm::l2Norm(velocity)) < 1.0;
+  if (result) {
+
+    if (not engineLimitReached) {
+      if (userSpeedLimitReached) {
+        logging::INFO("User speed (movePerSec_m) reached. It can be increased");
```

Comment:
Maybe "User speed (movePerSec_m) reached." is enough.

Typically, the user won't aim to go to the speed allowed by the max engine force but will rather adjust speed based on the characteristics that the simulated point cloud data should have. "It can be increased" may thus be confusing for the user.